src/nvme/types.h

```diff
@@ -3319,6 +3319,92 @@ struct nvme_media_unit_stat_log {
 	struct nvme_media_unit_stat_desc mus_desc[];
 };
 
+/**
+ * struct nvme_media_unit_config_desc 
+ * @muid: Media Unit Identifier
+ * @mudl: Media Unit Descriptor Length
+ * 
+ * Media Unit Configuration Descriptor
+ * Structure Definitions
+ */
+struct nvme_media_unit_config_desc {
+	__le16	muid;
+	__u8	rsvd2[4];
+	__le16	mudl;
+};
+
+/**
+ * struct nvme_channel_config_desc -
+ * @chanid:	Channel Identifier
+ * @chmus:	Number Channel Media Units
+ * 
+ * Channel Configuration Descriptor
+ * Structure Definitions
+ */
+struct nvme_channel_config_desc {
+	__le16	chanid;
+	__le16	chmus;
+	struct nvme_media_unit_config_desc mu_config_desc[];
+};
+
+/**
+ * struct nvme_end_grp_config_desc -
+ * @endgid:		Endurance Group Identifier
+ * @cap_adj_factor:	Capacity Adjustment Factor
+ * @tegcap:		Total Endurance Group Capacity
+ * @segcap:		Spare Endurance Group Capacity
+ * @end_est:		Endurance Estimate
+ * @egsets:		Number of NVM Sets
+ * @nvmsetid:		NVM Set Identifier
+ * @egchans:		Number of Channels
+ * 
+ * Endurance Group Configuration Descriptor
+ * Structure Definitions
+ */
+struct nvme_end_grp_config_desc {
+	__le16	endgid;
+	__le16	cap_adj_factor;
+	__u8	rsvd4[12];
+	__u8	tegcap[16];
+	__u8	segcap[16];
+	__u8	end_est[16];
+	__u8	rsvd64[16];
+	__le16	egsets;
+	__le16	nvmsetid[256];
```

Comment:
Is this member field fixed size to 256 bytes? The NVME spec I am looking at says it's variable size 
```
(EGSETS*2)+81 :
(EGSETS*2)+80
```
but maybe I am looking at an outdated spec (Revision 2.0a July 23rd, 2021)

Reply:
@igaw 
It's variable size as you mentioned. Maximum it can hold 65535. Will split the structure and incorporate variable size. Thanks!

Reply:
Alright, I was already wondering if I am looking at the wrong spec :)